internal/profile/profile.go

```diff
@@ -479,6 +488,54 @@ func EnhanceFromSchematic(
 	return prof, nil
 }
 
+// MergeOverlayProfile merges the overlay profile into the main profile.
+//
+// Only some fields from the overlay profile are merged into the main profile.
+func MergeOverlayProfile(
+	ctx context.Context,
+	artifactProducer ArtifactProducer,
+	prof *profile.Profile,
+	overlayRef artifacts.OverlayRef,
+) error {
+	overlayProfilePath, err := artifactProducer.GetOverlayArtifact(ctx, artifacts.Arch(prof.Arch), overlayRef, artifacts.OverlayKindProfiles)
+	if err != nil {
+		return fmt.Errorf("error getting overlay profiles %s: %w", overlayRef.TaggedReference, err)
+	}
+
+	var overlayProfile profile.Profile
+
+	overlayProfileData, err := os.ReadFile(filepath.Join(overlayProfilePath, overlayRef.Name+".yaml"))
+	if err != nil {
+		return fmt.Errorf("error reading overlay profile %s: %w", overlayProfilePath, err)
+	}
+
+	if err = yaml.Unmarshal(overlayProfileData, &overlayProfile); err != nil {
+		return fmt.Errorf("error unmarshalling overlay profile %s: %w", overlayProfilePath, err)
+	}
+
+	// zero out fields in the overlay profile which are set explicitly by the Image Factory
+	overlayProfile.Input = profile.Input{}
+	overlayProfile.Arch = ""
+	overlayProfile.Platform = ""
+	overlayProfile.Overlay = nil
+	overlayProfile.Version = ""
+	overlayProfile.SecureBoot = nil
+	overlayProfile.Output.Kind = profile.OutputKind(0)
```

Comment:
I think calling `outkind.<type>.String()` is more readable

Reply:
it's not a string here, but rather enum (int).

 I want a zero value here so that `merge.Merge` will skip it